### PR TITLE
DSD-1082: Header VQA fixes

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1082](https://jira.nypl.org/browse/DSD-1082)

## This PR does the following:

- Fixes three main issues found while doing VQA on the `Header` component:
  - Setting the width of the search form in the dropdown to a set max-width instead of being full-width.
Screen Shot 2022-08-03 at 4 54 46 PM

  - Increase sizes of icons in the "Log In" dropdown.
  - Decreases the outline of the desktop search submit button to 2px.

Other fixes:
- Better size of the NYPL logo on medium and large viewports, and pushes it up slightly above the upper nav.
Screen Shot 2022-08-03 at 4 51 53 PM

- Updates the "Donate" button in mobile to be all caps.
Screen Shot 2022-08-03 at 4 52 03 PM

- Adds a better border bottom to the desktop "Search" button to match the production Header's "Search" button.
Screen Shot 2022-08-03 at 4 51 09 PM

- Uses the proper color for the "Search" desktop button.
- Uses the proper color for the black backgrounds for mobile "Log In" and "Navigation" dropdowns.
Screen Shot 2022-08-03 at 4 55 15 PM

- Better alignment of the global alert notification text.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
Review Vercel URL: https://nypl-design-system-8ooc4je7y-nypl.vercel.app/iframe.html?args=&id=components-page-layout-header--header&viewMode=story

File: src/theme/components/header/header.ts [hunk 1]
```diff
@@ -11,7 +12,7 @@ export const headerFocus = {
   outlineColor: `${headerDarkBlue} !important`,
   outlineOffset: "0 !important",
   outlineStyle: "solid !important",
-  outlineWidth: "0.1875em !important",
+  outlineWidth: "0.25em !important",
 };
 
 const Header = {
```
Comment (bigfishdesign13) [inline src/theme/components/header/header.ts:18]: The `fontFamily` attribute below needs to be updated.  `’system ui’` should be changed to `’system-ui’`.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
I don’t know why I did not see this earlier, but there is a typo in the global `fontFamily` style.  Please see the comment that I left.  I see that you made a number of adjustments to font sizes and spacing.  I hope this update to the font family does not cause problems with those font style changes.

**Comment by EdwinGuzman** [comment]
The font family typo does make more changes needed! But better this way and I think now it does look closer to the actual NYPL Header.

**Review by bigfishdesign13 — COMMENTED** [review]
Do you want to ask Alkim to do a throrough round of VQA specifically looking at the fonts?  I can also help with that on Monday.

**Comment by EdwinGuzman** [comment]
Yes, let's ask Alkim. This ticket is already a `"qa-needed"` ticket so we'll assign it to him after this gets merged.

File: src/theme/components/header/headerLogin.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ const loginFocus = () => ({
 const HeaderLogin = {
   parts: ["logoutButton", "patronGreeting"],
   baseStyle: ({ patronName }) => ({
-    bg: { base: "ui.black", md: headerBlue },
+    bg: { base: headerBlack, md: headerBlue },
     boxShadow: "2px 2px 3px 4px rgb(100 100 100 / 25%)",
```
Comment (bigfishdesign13) [inline src/theme/components/header/headerLogin.ts]: This shadow should only be applied to the mobile view.

File: src/theme/components/header/headerLogin.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ const loginFocus = () => ({
...
     display: { base: "block", md: "flex" },
     flexDirection: "column",
```
Comment (bigfishdesign13) [inline src/theme/components/header/headerLogin.ts:16]: There is a `marginTop: { md: “1px” }` that is added on line 18.  Why was that added?  With that style in place, I am seeing a 1px gap between the dropdown and the Close button.

image

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Some very minor comments.

File: src/theme/components/header/headerLogin.ts (hunk 3 above)
Comment (EdwinGuzman) [reply src/theme/components/header/headerLogin.ts:16]: It's added because that's how it appears in the current NYPL Header.

Comment (bigfishdesign13) [reply src/theme/components/header/headerLogin.ts:16]: In Chrome, I do not see that same gap on NYPL.org.

image

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.  As noted in discussions outside github, the 1px gap in the `Log In` button is somethign that is already inconsistent in the current header based on browser, so we are not going to address that right now.
